src/mdrv2.cpp

```diff
@@ -117,12 +122,104 @@ void updatePcieInfo(UpdateDBusData infoPcieDevs) {
 
     for (auto& pciDev : pcieDevices)
     {
-        pciDev.second->pcieInfoUpdate();
+        pciDev.second->pcieInfoUpdate(motherboardPath);
+    }
+}
+
+void getMotherboardPath(sdbusplus::bus_t& bus, const std::string& inventoryPath,
+                        std::string& motherboardPath)
+{
+    // By default, look for System interface on any system/board/* object
+    std::string mapperAncestorPath = inventoryPath;
+    std::string matchParentPath = inventoryPath + "/board/";
+    bool requireExactMatch = false;
+
+    if (inventoryPath != defaultInventoryPath)
+    {
+        std::filesystem::path path(inventoryPath);
+
+        // Search under parent to find exact match for self
+        mapperAncestorPath = path.parent_path().string();
+        matchParentPath = mapperAncestorPath;
+        requireExactMatch = true;
+    }
+
+    auto method = bus.new_method_call(mapperBusName, mapperPath,
+                                      mapperInterface, "GetSubTreePaths");
+    method.append(mapperAncestorPath);
+    method.append(0);
+    method.append(std::vector<std::string>({systemInterface}));
+
+    try
+    {
+        std::vector<std::string> paths;
+        sdbusplus::message_t reply = bus.call(method);
+        reply.read(paths);
+
+        size_t pathsCount = paths.size();
+        for (size_t i = 0; i < pathsCount; ++i)
+        {
+            if (requireExactMatch && (paths[i] != inventoryPath))
+            {
+                continue;
+            }
+
+            motherboardPath = std::move(paths[i]);
+            break;
+        }
+        if (motherboardPath.empty())
+        {
+            phosphor::logging::log<phosphor::logging::level::ERR>(
+                "Failed to get system motherboard dbus path. Setting up a "
+                "match rule");
+
+            if (!motherboardConfigMatch)
```

Comment:
Was not able to test this branch.

Reply:
Wouldn't it be tested if you stop `entity-manager`? And repeat pcie transfer process?

Reply:
Tested as described in commit message